### PR TITLE
Add EEx.tokenize/2

### PR DESCRIPTION
**Pull request by feliperenan**

### CODE REVIEW

File: lib/eex/lib/eex.ex [hunk 1]
```diff
@@ -272,6 +272,35 @@ defmodule EEx do
...
+
+  ## Examples
+
+      iex> EEx.tokenize('foo', line: 1, column: 1)
+      {:ok, [{:text, 'foo', %{column: 1, line: 1}}, {:eof, %{column: 4, line: 1}}]}
+
+  """
+  @spec tokenize(contents :: binary | charlist, opts :: keyword) ::
+          {:ok, EEx.Tokenizer.token()} | {:error, String.t(), EEx.Tokenizer.metadata()}
+  def tokenize(contents, opts) do
```
Comment (feliperenan) [inline lib/eex/lib/eex.ex:293]:
```suggestion
  def tokenize(contents, opts \\ []) do
```

Maybe make it optional?

Comment (josevalim) [reply lib/eex/lib/eex.ex:294]: Yes!

let’s also document the tokens by bringing the specs from EEx.Tokenizer to this module!

**Review by cw789 — COMMENTED** [review]
@doc:
` * `column` - An integer to start as line. Default is 1.`

... `to start as column` ?

File: lib/eex/lib/eex/tokenizer.ex [hunk 2]
```diff
@@ -27,13 +17,6 @@ defmodule EEx.Tokenizer do
 
   Or `{:error, message, %{column: column, line: line}}` in case of errors.
   """
-  @spec tokenize(
```
Comment (feliperenan) [inline lib/eex/lib/eex/tokenizer.ex:30]: I guess we don't need it here anymore since we are documenting the public version of it, right?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**